worker/peergrouper/machinetracker_test.go

```diff
@@ -85,25 +89,7 @@ func (s *machineTrackerSuite) TestSelectMongoAddressReturnsErrNoSpaceMultipleClo
 		},
 	}
 
-	addr, err := m.SelectMongoAddress(666, spaceName)
-	c.Check(err, gc.ErrorMatches, `machine "3" has more than one non-local address and juju-ha-space is not set`)
-	c.Check(addr, gc.Equals, "")
-}
-
-func (s *machineTrackerSuite) TestSelectMongoAddressReturnsEmptyWhenNoAddressFound(c *gc.C) {
-	spaceName := network.SpaceName("")
-
-	m := &machineTracker{
-		id: "3",
-		addresses: []network.Address{
-			{
-				Value: "localhost",
-				Scope: network.ScopeMachineLocal,
-			},
-		},
-	}
-
-	addr, err := m.SelectMongoAddress(666, spaceName)
-	c.Assert(err, gc.IsNil)
-	c.Check(addr, gc.Equals, "")
+	addrs := network.HostPortsToStrings(m.GetPotentialMongoHostPorts(666))
+	sort.Strings(addrs)
+	c.Check(addrs, gc.DeepEquals, []string{"10.0.0.1:666", "192.168.5.5:666", "localhost:666"})
```

Comment:
this seems fairly bad that we would ever select 'localhost' along with other units. *Very* likely to just break things. Because the machine at 10.0.0.1 can't reach the machine that thinks of itself as 'localhost'.
We should be very cautious around code that could let this happen.

Reply:
This is a "dumb" method in that it does no discrimination; the real selection happens upstream.
I did it this way to preserve the old logging behaviour of selected x from x, y, z.
The test makes it look like a potential error path, but I will strive toward something more cogent.

Reply:
I can understand that there may not actually be an issue here. I would just structure the text so that other developers looking at this don't think that we would want 'localhost' to be valid.
Maybe that's just updating the test so the initial config isn't using localhost.